development/build/scripts.js

```diff
@@ -0,0 +1,361 @@
+const materialUIDependencies = ['@material-ui/core']
+const reactDepenendencies = dependencies.filter((dep) => dep.match(/react/))
+const d3Dependencies = ['c3', 'd3']
+
+const externalDependenciesMap = {
+  background: [
+    '3box',
+  ],
+  ui: [
+    ...materialUIDependencies, ...reactDepenendencies, ...d3Dependencies,
+  ],
+}
+
+function createScriptTasks ({ browserPlatforms, livereload }) {
+
+  // internal tasks
+  const core = {
+    // dev tasks (live reload)
+    dev: createTasksForBuildJsExtension({ taskPrefix: 'scripts:core:dev', devMode: true }),
+    testDev: createTasksForBuildJsExtension({ taskPrefix: 'scripts:core:test-live', devMode: true, testing: true }),
+    // built for CI tests
+    test: createTasksForBuildJsExtension({ taskPrefix: 'scripts:core:test', testing: true }),
+    // production
+    prod: createTasksForBuildJsExtension({ taskPrefix: 'scripts:core:prod' }),
+  }
+  const deps = {
+    background: createTasksForBuildJsDeps({ filename: 'bg-libs', key: 'background' }),
+    ui: createTasksForBuildJsDeps({ filename: 'ui-libs', key: 'ui' }),
+  }
+
+  // high level tasks
+
+  const prod = taskParallel(
+    deps.background,
+    deps.ui,
+    core.prod,
+  )
+
+  const dev = core.dev
+  const testDev = core.testDev
+
+  const test = taskParallel(
+    deps.background,
+    deps.ui,
+    core.test,
+  )
+
+  return { prod, dev, testDev, test }
+
+
+  function createTasksForBuildJsDeps ({ key, filename }) {
+    return createTask(`scripts:deps:${key}`, bundleTask({
+      label: filename,
+      filename: `${filename}.js`,
+      buildLib: true,
+      dependenciesToBundle: externalDependenciesMap[key],
+      devMode: false,
+    }))
+  }
+
+
+  function createTasksForBuildJsExtension ({ taskPrefix, devMode, testing }) {
+    const standardBundles = [
+      'background',
+      'ui',
+      'phishing-detect',
+    ]
+
+    const standardSubtasks = standardBundles.map((filename) => {
+      return createTask(`${taskPrefix}:${filename}`,
+        createBundleTaskForBuildJsExtensionNormal({ filename, devMode, testing })
+      )
+    })
+    // inpage must be built before contentscript
+    // because inpage bundle result is included inside contentscript
+    const contentscriptSubtask = createTask(`${taskPrefix}:contentscript`,
+      createTaskForBuildJsExtensionContentscript({ devMode, testing })
+    )
+
+    // task for initiating livereload
+    const initiateLiveReload = () => {
+      if (devMode) {
+        // trigger live reload when the bundles are updated
+        // this is not ideal, but overcomes the limitations:
+        // - run from the main process (not child process tasks)
+        // - after the first build has completed (thus the timeout)
+        // - build tasks never "complete" when run with livereload + child process
+        setTimeout(() => {
+          watch('./dist/*/*.js', (event) => {
+            livereload.changed(event.path)
+          })
+        }, 75e3)
```

Comment:
does this work? did you try it? if so what did you QA?

Reply:
Yep - it seemed to work! I tried building, then changing some JavaScript files, both during the build and afterwards.

The reason it works is that the initial build seems to consistently emit an `add` event, then a `changed` event, in rapid succession. Then any subsequent updates are `change` events (I figured we could safely ignore the possibility of new files being added or files being removed during typical development)

Reply:
I can't get this to work; I don't know why the `change` event is emitted sometimes but not always. The original solution will have to do - we can increase the time if need be.

Reply:
seems the glob is evaluated at start, and not ongoing
it might work in the case where it runs before clean completes? so it watches for the files even if they are deleted

yeah totally agree its not ideal but it works for now